packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -335,29 +438,55 @@ class _IOSDeviceLogReader extends DeviceLogReader {
 }
 
 class _IOSDevicePortForwarder extends DevicePortForwarder {
-  _IOSDevicePortForwarder(this.device);
+  _IOSDevicePortForwarder(this.device) : _forwardedPorts = new List<ForwardedPort>();
 
   final IOSDevice device;
 
+  final List<ForwardedPort> _forwardedPorts;
+
   @override
-  List<ForwardedPort> get forwardedPorts {
-    final List<ForwardedPort> ports = <ForwardedPort>[];
-    // TODO(chinmaygarde): Implement.
-    return ports;
-  }
+  List<ForwardedPort> get forwardedPorts => _forwardedPorts;
 
   @override
   Future<int> forward(int devicePort, {int hostPort: null}) async {
     if ((hostPort == null) || (hostPort == 0)) {
       // Auto select host port.
       hostPort = await findAvailablePort();
     }
-    // TODO(chinmaygarde): Implement.
-    return hostPort;
+
+    // Usage: iproxy LOCAL_TCP_PORT DEVICE_TCP_PORT UDID
+    Process process = await runCommand(<String>[
+      device.iproxyPath,
+      hostPort.toString(),
+      devicePort.toString(),
+      device.id,
+    ]);
+
+    ForwardedPort forwardedPort = new ForwardedPort.withContext(hostPort,
+        devicePort, process);
+
+    printTrace("Forwarded port $forwardedPort");
+
+    _forwardedPorts.add(forwardedPort);
+
+    return 1;
   }
 
   @override
   Future<Null> unforward(ForwardedPort forwardedPort) async {
-    // TODO(chinmaygarde): Implement.
+    if (!_forwardedPorts.remove(forwardedPort)) {
+      // Not in list. Nothing to remove.
+      return null;
+    }
+
+    printTrace("Unforwarding port $forwardedPort");
+
+    Process process = forwardedPort.context;
+
+    assert(process != null);
```

Comment:
This is a rather paranoid check since instances of the forwarded port are never created outside that class. I initially wanted to assert with a message but it seemed oddly difficult to do. Will add just a log instead.